crates/rpc/rpc/src/otterscan.rs

```diff
@@ -76,6 +79,7 @@ where
         > + EthApiTypes<
             NetworkTypes: Network<
                 TransactionResponse = WithOtherFields<reth_rpc_types::Transaction>,
+                ReceiptResponse = AnyTransactionReceipt,
```

Comment:
this constraint can now be removed since https://github.com/alloy-rs/alloy/issues/1245 is closed. the new trait methods on `alloy_network_primitives::ReceiptResponse` should be enough to access fields on the receipt throughout this `OtterscanServer` impl.